src/main/java/net/bramp/db_patterns/locks/MySQLSleepBasedCondition.java

```diff
@@ -89,13 +89,14 @@ protected boolean awaitNanosInternal(long nanosTimeout) throws InterruptedExcept
 
 					// Adjust nanosTimeout (due to time it took to get a connection)
 					nanosTimeout -= (System.nanoTime() - now);
-					if (nanosTimeout <= 0) {
-					    LOG.trace("After connection acquired, timeout was negative or zero: {}", nanosTimeout);
+					float timeout = Math.round(nanosTimeout / 1000000.0) / 1000f;
+					if (timeout <= 0) {
+					    LOG.trace("After connection acquired, timeout was negative or zero: {}", timeout);
 					    return false;
 					}
 
 					// Convert to seconds, but round to whole number of milliseconds
```

Comment:
The rounded timeout (the one that will be given to SLEEP()) is negative or zero. Should I be more precise, like "After connection acquired and timeout rounded to whole number of milliseconds, the timeout was negative or zero."?